Index: RegisterContextMach_x86_64.cpp
===================================================================
--- RegisterContextMach_x86_64.cpp	(revision 136766)
+++ RegisterContextMach_x86_64.cpp	(working copy)
@@ -7,6 +7,7 @@
 //
 //===----------------------------------------------------------------------===//
 
+#if defined(__APPLE__)
 
 // C Includes
 #include <mach/thread_act.h>
@@ -68,3 +69,4 @@
     return ::thread_set_state(tid, flavor, (thread_state_t)&exc, EXCWordCount);
 }
 
+#endif
Index: RegisterContextMach_arm.cpp
===================================================================
--- RegisterContextMach_arm.cpp	(revision 136766)
+++ RegisterContextMach_arm.cpp	(working copy)
@@ -7,6 +7,8 @@
 //
 //===----------------------------------------------------------------------===//
 
+#if defined(__APPLE__)
+
 #include "RegisterContextMach_arm.h"
 
 // C Includes
@@ -82,4 +84,4 @@
     return ::thread_set_state(tid, flavor, (thread_state_t)&dbg, DBGWordCount);
 }
 
-
+#endif
Index: RegisterContextMach_i386.cpp
===================================================================
--- RegisterContextMach_i386.cpp	(revision 136766)
+++ RegisterContextMach_i386.cpp	(working copy)
@@ -7,6 +7,7 @@
 //
 //===----------------------------------------------------------------------===//
 
+#if defined(__APPLE__)
 
 // C Includes
 #include <mach/thread_act.h>
@@ -68,4 +69,4 @@
     return ::thread_set_state(tid, flavor, (thread_state_t)&exc, EXCWordCount);
 }
 
-
+#endif
